src/stratigraphy/util/predictions.py

```diff
@@ -266,8 +260,40 @@ def evaluate_metadata_extraction(self, ground_truth_path: Path) -> OverallBoreho
         Args:
             ground_truth_path (Path): The path to the ground truth file.
         """
-        metadata_per_file = BoreholeMetadataList()
+        metadata_per_file: BoreholeMetadataList = BoreholeMetadataList()
 
         for file_prediction in self.file_predictions_list:
             metadata_per_file.add_metadata(file_prediction.metadata)
         return MetadataEvaluator(metadata_per_file, ground_truth_path).evaluate()
+
+    @classmethod
+    def from_json(cls, prediction_from_file: dict) -> "OverallFilePredictions":
+        """Converts a dictionary to an object.
+
+        Args:
+            prediction_from_file (dict): A dictionary representing the predictions.
+
+        Returns:
+            OverallFilePredictions: The object.
+        """
+        overall_file_predictions = OverallFilePredictions()
+        for file_name, file_data in prediction_from_file.items():
+            metadata = BoreholeMetadata.from_json(file_data.get(file_name)["metadata"], file_name)
```

Comment:
I would expect that you can just write `file_data[""metadata]` here, but we seem to have some "double nesting" of the filename in predictions.json. Do you know where this comes from?

![image](https://github.com/user-attachments/assets/e7dcd310-d769-40e4-b11b-4b55566fbddf)

Reply:
I do believe that this section of the code was not used in the end. However, you are right the `to_json` method of the `OverallFilePredictions` object was generating double-nested json files. I addressed this and also wrote some tests to ensure that the output format does not change unintentionally over time.